src/webgpu/api/operation/resource_init/depth_stencil_attachment_clear.spec.ts

```diff
@@ -0,0 +1,242 @@
+          }
+          `,
+        }),
+      },
+      fragmentStage: {
+        entryPoint: 'main',
+        module: this.makeShaderModule('fragment', {
+          glsl: `#version 310 es
+          precision highp float;
+          layout(location = 0) out float outSuccess;
+
+          void main() {
+            outSuccess = 1.0;
+          }
+          `,
+        }),
+      },
+      colorStates: [
+        {
+          format: 'r8unorm',
+        },
+      ],
+      depthStencilState: {
+        format,
+        stencilFront: {
+          compare: 'equal',
+        },
+        stencilBack: {
+          compare: 'equal',
+        },
+      },
+      primitiveTopology: 'triangle-list',
+      sampleCount,
+    });
+    this.stencilTestPipelineCache.set(key, pipeline);
+    return pipeline;
+  }
+
+  checkContents(
+    texture: GPUTexture,
+    state: InitializedState,
+    subresourceRange: SubresourceRange
+  ): void {
+    for (const viewDescriptor of this.generateTextureViewDescriptorsForRendering(
+      this.params.aspect,
+      subresourceRange
+    )) {
+      const width = this.textureWidth >> viewDescriptor.baseMipLevel!;
+      const height = this.textureHeight >> viewDescriptor.baseMipLevel!;
+
+      const renderTexture = this.device.createTexture({
+        size: [width, height, 1],
+        format: 'r8unorm',
+        usage: C.TextureUsage.OutputAttachment | C.TextureUsage.CopySrc,
+        sampleCount: this.params.sampleCount,
+      });
+
+      let resolveTexture = undefined;
+      let resolveTarget = undefined;
+      if (this.params.sampleCount > 1) {
+        resolveTexture = this.device.createTexture({
+          size: [width, height, 1],
+          format: 'r8unorm',
+          usage: C.TextureUsage.OutputAttachment | C.TextureUsage.CopySrc,
+        });
+        resolveTarget = resolveTexture.createView();
+      }
+
+      const commandEncoder = this.device.createCommandEncoder();
+      const pass = commandEncoder.beginRenderPass({
+        colorAttachments: [
+          {
+            attachment: renderTexture.createView(),
+            resolveTarget,
+            loadValue: [0, 0, 0, 0],
+            storeOp: 'store',
+          },
+        ],
+        depthStencilAttachment: {
+          attachment: texture.createView(viewDescriptor),
+          depthStoreOp: 'store',
+          depthLoadValue: 'load',
+          stencilStoreOp: 'store',
+          stencilLoadValue: 'load',
+        },
+      });
+
+      switch (this.params.readMethod) {
+        case ReadMethod.DepthTest:
+          pass.setPipeline(
+            this.getDepthTestReadbackPipeline(state, this.params.format, this.params.sampleCount)
+          );
+          break;
+
+        case ReadMethod.StencilTest:
+          pass.setPipeline(
+            this.getStencilTestReadbackPipeline(this.params.format, this.params.sampleCount)
+          );
+          pass.setStencilReference(initializedStateAsStencil(state));
```

Comment:
add a comment here too, since this is critical to the test

Reply:
Done